book/src/fundamentals/zkbook_commitment.md

```diff
@@ -25,40 +25,133 @@ $$
+
+$$
+\begin{aligned}
+A + B &= \mathsf{commit}(x_a, r_a) + \mathsf{commit}(x_b, r_b) \\
+&= x_aG +r_aH + x_bG + r_bH \\
+&= (x_a + x_b)G + (r_a + r_b)H \\
+&= \mathsf{commit}(x_a + x_b, r_a + r_b)
+\end{aligned}
+$$
+
+In other words, the sum of commitments $A$ and $B$ is equal to the commitment of the sum of the two committed values $x_a$ and $x_b$ and blinders $r_a$ and $r_b$.
+
+> As a cryptographic primitive, the ability to find a public curve point $H$ for which no one knows the discrete logarithm may, at first, seem rather mind-blowing and powerful.
+>
+> Actually, it's as easy as it is awesome to find such a point--- simply perform rejection sampling by cryptographically hashing $G$ (or, respectively, the
+> hash output), using the output as the $x$-coordinate of a candidate point on $E$ and checking whether it's valid.  The first valid curve point obtained is $H$ and
+> by the hardness assumption of the elliptic curve discrete logarithm problem (ECDLP), no one knows it.
+>
+> Since approximately half of the hash outputs will be valid curve points on $E$, sampling will terminate very quickly.  Indeed, as we will see later,
+> this process can be used to sample many public curve points $G_1, \ldots, G_n$ for which the discrete logarithms are unknown; the so-called *hash to curve* algorithm.
+
+## Pedersen commitments
+
+The homomorphic commitment $\mathsf{commit}(x, r) = xG + rH$ described above is known as a Pedersen commitment. If you remove the $rH$ term you
+get a non-hiding commitment, called a *Pedersen hash*. Both rely on the ECDLP hardness assumption.
+
+This means that, at least theoretically, you might be lucky (or have a quantum computer) and figure out that $H = hG$, which would allow you to find different values $x'$ and $h'$ to open the commitment.
+We say that pedersen commitments are **computationally binding** and not unconditionally binding.  For example, you could express $c = xG + rH$ alternatively as
+$c = xG + rh G = (x + rh) G$ and compute a satisfying opening pair $x' = xh$ and $r' = \frac{x}{h}$.
+
+On the other hand, Pedersen commitments are **unconditionally hiding**, as there is no way (even with a magic computer) to reveal what $x$ is without knowing $r$.
+Lack of perfect binding is the reason why most of the "proofs" we will see later in this book are not referred to as proofs, but instead are referred to as
+**arguments** of knowledge (although we may care little about this distinction). Just remember that you need perfect binding to be called a proof.
+
+> Interestingly, it is impossible to have a commitment scheme that has both perfect hiding and perfect binding.
+
+To recap, in cryptography the following distinctions are important
+
+- **Perfect.** The property that an algorithm is statistically sound without hardness assumptions, also known as unconditional or statistical soundness.
```

Comment:
cc @imeckler that's what we were talking about the other day. perfect = statistical